src/teacher/forms.py

```diff
@@ -10,7 +12,17 @@ class Meta:
 
 
 class TeacherAddForm(TeacherBaseForm):
-    pass
+
+    def clean(self):
+        cleaned_data = super().clean()
+        phone_number = cleaned_data.get('phone_number').strip()
+        email = cleaned_data.get('email').strip()
+
+        is_teacher_exists = Teacher.objects.filter(Q(phone_number=phone_number) |
+                                                   Q(email=email)).exists()
+        if is_teacher_exists:
+            error_massage = "Teacher not added. Teacher with such phone_number or email is exists! Try again:"
+            raise ValidationError(error_massage)
```

Comment:
Consider returning `cleaned_date` here. For example, you might want to derive from this form and add some additional validation in derived class. Example: `FlatpageForm` from `django.auth.contrib.flatpages`